python/seldon_core/seldon_methods.py

```diff
@@ -29,6 +30,27 @@
 logger = logging.getLogger(__name__)
 
 
+def handle_raw_custom_metrics(
+    msg: Union[prediction_pb2.SeldonMessage, Dict],
+    seldon_metrics: SeldonMetrics,
+    is_proto: bool,
+):
+    """
+    Update SeldonMetrics object with custom metrics from raw methods.
+    If INCLUDE_METRICS_IN_CLIENT_RESPONSE environmental variable is set to "true"
+    metrics will be dropped from msg.
+    """
+    if is_proto:
+        metrics = seldon_message_to_json(msg.meta).get("metrics", [])
+        if metrics and not return_metrics_to_client():
+            del msg.meta.metrics[:]
```

Comment:
I assume this operation is quite fast , but just wondering what is the reason to prefer `del` vs `...metrics = None`?

Reply:
`del` was the first thing I found to work on Protobuf. I think I tried setting `...metrics = []` and it didn't work. Will setting `None` work?

Reply:
Got it, ok, the reason why I ask is same as above, but tbh even if it was to iterate across all the elements within custom metrics it would not be inefficient. I just had a look and apparently there is a native proto `Clear()` method, which may also do the trick https://developers.google.com/protocol-buffers/docs/pythontutorial#standard-message-methods

Reply:
` ...metrics = None` would raise
```python
AttributeError: Assignment not allowed to repeated field "metrics" in protocol message object.
```